core/src/main/scripts/importer/importOncokbDiscreteCNA.py

```diff
@@ -0,0 +1,288 @@
+    meta_dict = libImportOncokb.read_meta_file(meta_cna_file_path)
+    if 'pd_annotations_filename' in meta_dict:
+        raise RuntimeError(
+            "Custom driver annotations filename already specified in discrete CNA meta file. Please remove and rerun.")
+    if path.exists(pd_file_path):
+        raise RuntimeError(
+            "Custom driver annotations file '" + pd_file_path + "' for discrete CNA already exists . Please remove and rerun.")
+    check_required_columns(libImportOncokb.get_first_line(open_cna_file(cna_file_path)).rstrip('\n').split('\t'))
+
+    global portal_instance
+    if hasattr(args, 'portal_info_dir') and args.portal_info_dir is not None:
+        portal_instance = validateData.load_portal_info(args.portal_info_dir, logger,
+                                           offline=True)
+    else:
+        portal_instance = validateData.load_portal_info(server_url, logger)
+
+    features = get_features(cna_file_path)
+    id_to_annotation = fetch_oncokb_annotations(features)
+    for feature in features:
+        if feature['id'] in id_to_annotation:
+            feature['oncogenic'] = id_to_annotation[feature['id']]['oncogenic']
+
+    print("Updating study files ...", end = '')
+    write_annotations_to_file(features, pd_file_path)
+    update_cna_metafile(meta_cna_file_path, pd_file_name)
+    print(" DONE")
+
+    logger.info('Import complete')
+
+    return exit_status_handler.get_exit_status()
+
+
+def open_cna_file(file_name):
+    """Open CNA file and handle exception when not found."""
+    try:
+        file = open(file_name)
+    except FileNotFoundError:
+        raise FilenotFoundError("Could not open discrete CNA file at path '" + file_name + "'")
+    return file
+
+
+def get_features(cna_file_path):
+    """Extract CNA events from CNA data file."""
+    header_elements = libImportOncokb.get_first_line(open_cna_file(cna_file_path)).rstrip('\n').split('\t')
+    header_indexes = {}
+    for required_column in required_cna_columns + ['Entrez_Gene_Id']:
+        header_indexes[required_column] = header_elements.index(required_column)
+    sample_ids = [i for j, i in enumerate(header_elements) if j not in header_indexes.values()]
+    sample_indexes = {}
+    for sample_id in sample_ids:
+        sample_indexes[sample_id] = header_elements.index(sample_id)
+
+    features = []
+    cna_file = open_cna_file(cna_file_path)
+    print("Reading features from file ...", end = '')
+    for line in cna_file:
+        if line == '\n' or line.startswith('#') or line.startswith(header_elements[0]):
+            continue  # skip comment and header line
+        line_elements = line.rstrip('\n').split('\t')
+        for sample_id in sample_ids:
+            feature = {}
+            feature['sample_id'] = sample_id
+            feature['alteration'] = int(line_elements[sample_indexes[sample_id]])
+            # cna value 0 (no CNA) is skipped
+            if feature['alteration'] == 0:
```

Comment:
Shouldn't you skip all values that not in cna_alteration_types.values()?

Reply:
No, the responsibility of this script is to import oncokb annotations. When CNA != 0 means there is a CNA event and this script imports the annotation. The selective import AMP and HOMDEL is a responsibility of the java import code. We should not distribute this logic elsewhere.

Reply:
Yes, good catch!

Reply:
Updated